projects/ghostscript/gstoraster_fuzzer.cc

```diff
@@ -110,7 +111,39 @@ static int gs_to_raster_fuzz(const unsigned char *buf, size_t size)
 	return 0;
 }
 
+#ifdef PDF_TARGET
+// Returns 1 if this has a valid PDF header and 0 otherwise
+static int quick_check_pdf(const uint8_t *data, size_t size) {
+	// PDF checks. Exit early if we don't have a valid PDF signature.
+	if (size < 5) {
+		return 0;
+	}
+
+	// Check PDF tag. We do this because we want to use seeds
+	if (data[0] != 0x25 || data[1] != 0x50 || data[2] != 0x44 || data[3] != 0x46 || data[4] != 0x2d) {
+		return 0;
+	}
+	return 1;
+}
+#endif
+
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
-	gs_to_raster_fuzz(data, size);
+#ifdef MULTIPLE_COLORS
```

Comment:
Why not just make another file?  gs_to_raster_fuzz is basically a library function anyway.

Reply:
No strong reason I didn't. The macro seemed easier. However, I can split it if you prefer -- I can separate gs_to_raster_fuzz into a separate file and make each of the three fuzzers into small stubs. Do you prefer it this way?

Reply:
Yes please.  It's just easier to reason about without the macros.  It'll be easier to change in the future as well.